core/lowering/lowering.cpp

```diff
@@ -48,18 +48,26 @@ int AutocastLongInputs(
       auto dtype = dtype_input->second.value();
       // Currently, we do not autocast inputs for which the determined type is not long
       if (dtype != at::kLong) {
+        LOG_DEBUG(
+            "Skipping autocast for tensor " << input->debugName() << ", since its dtype is " << dtype
+                                            << " and not at::kLong");
         continue;
       }
 
       LOG_DEBUG("Inserting aten::to casting " << input->debugName() << " to dtype " << dtype);
 
       // Generate cast node sending input tensors to the inferred or specified datatype (long)
+      torch::jit::Value *const_false, *cuda, *none_val;
+      if (num_autocasts == 0) {
+        // Only generate constants once and reuse for all autocasts
+        const_false = g->insertConstant(0);
+        const_false->setType(torch::jit::BoolType::get());
+        cuda = g->insertConstant(target_device_name);
+        cuda->setType(torch::jit::DeviceObjType::get());
+        none_val = g->insertNode(g->createNone())->output();
+      }
```

Comment:
Shouldn't this block of code be at line 33 before the for loop begins in order to be reused ?

Reply:
I don't think this should be outside of the `for` loop, because we only want to insert it if the checks on lines 39, 43, and 50 pass, in which case an autocast will be inserted. The code already increments the `num_autocasts` variable when a cast is inserted (line 80), so this code should only run maximally once. What are your thoughts on this?

Reply:
Oh got it. I was confused a bit. Makes sense